testbench/common.py

```diff
@@ -662,8 +697,11 @@ def streamer():
                     yield d[r:chunk_end]
                 if bytes_yield >= limit:
                     database.dequeue_next_instruction(test_id, method)
-                    if conn is not None:
-                        conn.close()
+                    # Accessing request object in unit test will give "working out of request context" exception. Ignoring these exception.
```

Comment:
The answer to that may be to call `_make_closer()` in the request context, and pass the result (the created "closer" function) to `__get_streamer_response_fn()`.